r/tests/testthat/test-dplyr.R

```diff
@@ -1035,10 +1035,20 @@ test_that("log functions", {
     df
   )
 
-  expect_error(
-    nse_funcs$log(Expression$scalar(x), base = 5),
-    "`base` values other than exp(1), 2 and 10 not supported by Arrow",
-    fixed = TRUE
+  expect_dplyr_equal(
+    input %>%
+      mutate(y = log(x, base = 5)) %>%
+      collect(),
+    df
+  )
+
+  expect_dplyr_equal(
+    input %>%
+      # suppress 'NaNs produced' warning on the first row of df
+      # that evaluates to NaN (R raises warning but Arrow does not)
+      suppressWarnings(mutate(., y = log(x, base = x))) %>%
```

Comment:
Looks like the old Windows build failed--perhaps the magrittr improvement that supports this is too new for 3.6 binary packages. So maybe the best option is to move the suppressWarnings and the accompanying comment to be around/outside of expect_dplyr_equal().

Reply:
log(x) with base 1 is not a reasonable calculation because this implies solving the following equation `1^y = x` which is only defined for `x = 1`. Nevertheless, most math libraries will yield a value of infinity for base 1 logarithms regardless of `x`. Arrow C++ follows this scheme.

Reply:
I think it is reasonable. I also noticed that these corner cases are not tested in the C++ log tests, but that corresponds to another PR.